insecureinc/src/main/webapp/cwe20loggedin.jsp

```diff
@@ -0,0 +1,93 @@
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
+<%@ page import="inc.insecure.*" %>
+<%@ page import="insecure.inc.Constants" %>
+<%
+
+String alertVisibility = "hidden";
+String error = "";
+String logoutParameter = request.getParameter("logout");
+
+if(logoutParameter!=null){
+  if(logoutParameter.equals("true")){
+    response.sendRedirect("cwe20.jsp?loggedin=false");	
+  }
+}
+
+if(session == null || session.getAttribute("cwe20loggedin") == null || !(boolean)session.getAttribute("cwe20loggedin")) {
+	response.sendRedirect("cwe20.jsp?loggedin=false");
+}
+else {
+  String cost = request.getParameter("cost");
+  int costParsed = 0;
+
+  try {
+    costParsed = Integer.parseInt(cost);
+  } catch (Exception e) {
+    cost = null;
+  }
+
+  if(cost != null) {
+    alertVisibility = "";
+
+    if(costParsed == 0) {
+      error = "The subscription amount cannot be zero!";
+    } else if (costParsed > 0) {
```

Comment:
@timmyteo did you mean costParsed == 60 here? (the participant passes the challenge if they are able to alter the 60 value that is embedded in the html on the else branch)